internal/fnruntime/container.go

```diff
@@ -112,7 +132,36 @@ func (f *ContainerFn) Run(reader io.Reader, writer io.Writer) error {
 	return nil
 }
 
-func (f *ContainerFn) getDockerCmd() (*exec.Cmd, context.CancelFunc) {
+func (f *ContainerFn) runPodman(reader io.Reader, writer io.Writer) error {
+	errSink := bytes.Buffer{}
+	cmd, cancel := f.getCmd(podmanBin)
+	defer cancel()
+	cmd.Stdin = reader
+	cmd.Stdout = writer
+	cmd.Stderr = &errSink
+
+	if err := cmd.Run(); err != nil {
+		var exitErr *exec.ExitError
+		if goerrors.As(err, &exitErr) {
+			return &ExecError{
+				OriginalErr:    exitErr,
+				ExitCode:       exitErr.ExitCode(),
+				Stderr:         filterPodmanCLIOutput(&errSink),
+				TruncateOutput: printer.TruncateOutput,
+			}
+		}
+		return fmt.Errorf("unexpected function error: %w", err)
+	}
+
+	if errSink.Len() > 0 {
+		f.FnResult.Stderr = filterPodmanCLIOutput(&errSink)
+	}
+	return nil
+}
```

Comment:
We are repeating this code for both. Wondering if there is a better way to structure this code.

Reply:
Updated